task/deprecated-image-check/0.1/deprecated-image-check.yaml

```diff
@@ -33,62 +30,66 @@ spec:
       image: registry.access.redhat.com/ubi8/ubi
       script: |
         #!/usr/bin/env bash
-        echo "Querying Pyxis for $(params.IMAGE_REPOSITORY)..."
-        http_code=$(curl -s -k -o $(workspaces.sanity-ws.path)/repository_data.json -w '%{http_code}' "https://catalog.redhat.com/api/containers/v1/repositories/registry/$(params.IMAGE_REGISTRY)/repository/$(params.IMAGE_REPOSITORY)")
-
-        echo "Response code: $http_code"
-        echo -n $http_code > $(results.PYXIS_HTTP_CODE.path)
+        export IMAGE_REPO_PATH=$(workspaces.sanity-ws.path)/${IMAGE_REPOSITORY}
+        IFS=$'\n' read -d'' -r -a IMAGE_ARRAY <<(echo -n "$(params.BASE_IMAGES_DIGESTS)" | sed 's/\\n/\'$'\n''/g' && printf '\0')
+        for BASE_IMAGE in ${IMAGE_ARRAY[@]};
+        do
+          IFS=:'/' read -r IMAGE_REGISTRY IMAGE_WITH_TAG <<< $BASE_IMAGE; echo "[$IMAGE_REGISTRY] [$IMAGE_WITH_TAG]"
+          IMAGE_REPOSITORY=`echo $IMAGE_WITH_TAG | cut -d ":" -f1`
+          IMAGE_REGISTRY=${IMAGE_REGISTRY//registry.redhat.io/registry.access.redhat.com}
+          mkdir -p ${IMAGE_REPO_PATH}
+          echo "Querying Pyxis for $BASE_IMAGE..."
+          http_code=$(curl -s -k -o ${IMAGE_REPO_PATH}/repository_data.json -w '%{http_code}' "https://catalog.redhat.com/api/containers/v1/repositories/registry/registry.access.redhat.com/repository/rhscl/nodejs-8-rhel7")
```

Comment:
The `https://catalog.redhat.com/api/containers/v1/repositories/registry/registry.access.redhat.com/repository/rhscl/nodejs-8-rhel7` has hardcoded values pointing to the deprecated `rhscl/nodejs-8-rhel7` repo, I think this is the cause for the failing e2e tests.
The url should be using the IMAGE_REGISTRY and IMAGE_REPOSITORY variables, something like this:
```
https://catalog.redhat.com/api/containers/v1/repositories/registry/${IMAGE_REGISTRY}/repository/${IMAGE_REPOSITORY}
```

Reply:
Thanks for the update @srivickynesh it looks good now.